skrifa/src/outline/glyf/hint/value_stack.rs

```diff
@@ -112,6 +112,24 @@ impl<'a> ValueStack<'a> {
         Ok(self.pop()? as usize)
     }
 
+    /// Convenience method for popping a value intended as a count.
+    ///
+    /// When a negative value is encountered, returns an error in pedantic mode
+    /// or 0 otherwise.
+    pub fn pop_count_checked(&mut self) -> Result<usize, HintErrorKind> {
+        let value = self.pop()?;
+        if value < 0 {
+            if self.is_pedantic {
+                Err(HintErrorKind::InvalidStackValue(value))
+            } else {
+                Ok(0)
+            }
+        } else {
+            // Positive i32 always fits into usize
+            Ok(value as usize)
+        }
+    }
```

Comment:
i'd probably have written,

```rust
if value < 0 && self.is_pedantic {
    // err path
} else {
    Ok(value.max(0) as usize)
}
```

but of course your version is totally fine

Reply:
I like your version better, thanks!